### PR TITLE
Support default timeout for API calls.

### PR DESCRIPTION
**Pull request by alexjurkiewicz**
Default timeout of 5 minutes. Let timeout exceptions from requests
bubble up, rather than wrapping them.

### CODE REVIEW

**Comment by alexjurkiewicz** [comment]
fixed tests

**Comment by alexjurkiewicz** [comment]
ping - are you interested in this?

**Comment by os** [comment]
Thanks @alexjurkiewicz. Sorry, I haven't had a chance to look at this yet. I'll try to spare some time this week.

**Comment by diwu1989** [comment]
300s is a very long default timeout
The API generally returns in 0.5s to 1s for me, maybe something more like a 10s default is more reasonable?

**Comment by os** [comment]
Thanks, @alexjurkiewicz, looks good. There is only a couple of things:

* `timeout` parameter should be optional. It is being passed as an optional parameter on `Slacker`. However, it is defined as a required parameter for both `BaseAPI` and `IncomingWebhook`.
* `timeout` parameter should come after `token`. This would still work for most cases, but it's inconsistent and more importantly, changing the function signature that could break for the others. Imagine they have a line of code that initialise `Users` class with a positional argument like `User(token)`. In that case it would break because it would set the `timeout`, not the `token` value. We just need to be careful about backwards compatibility.

**Comment by os** [comment]
@diwu1989 @alexjurkiewicz Ten seconds sounds reasonable to me.

File: slacker/__init__.py [hunk 1]
```diff
@@ -408,8 +411,9 @@ def access(self, client_id, client_secret, code, redirect_uri=None):
 
 
 class IncomingWebhook(object):
-    def __init__(self, url=None):
+    def __init__(self, timeout, url=None):
```
Comment (diwu1989) [inline slacker/__init__.py]: should be
```py
def __init__(self, url=None, timeout=DEFAULT_TIMEOUT):

File: slacker/__init__.py [hunk 2]
```diff
@@ -40,7 +41,8 @@ def __init__(self, body):
 
 
 class BaseAPI(object):
-    def __init__(self, token=None):
+    def __init__(self, timeout, token=None):
```
Comment (diwu1989) [inline slacker/__init__.py]:
```py
def __init__(self, token=None, timeout=DEFAULT_TIMEOUT):

**Comment by alexjurkiewicz** [comment]
Cheers, I'll submit all these fixes next week

**Comment by alexjurkiewicz** [comment]
Updated as per feedback!

**Comment by os** [comment]
@alexjurkiewicz Thanks, I'll look into that sometime this week.

**Comment by alexjurkiewicz** [comment]
Nice!

**Comment by os** [comment]
@alexjurkiewicz This is a big improvement. Thank you!

**Pull request merged**